### DISCUSSION TITLE
Extension of the "brute force" parser for the recipe ingredients.

### DISCUSSION THREAD

**Discussion by blueline13** [mealie-recipes/mealie#1129] [Feature Request]
I have a question or suggestion about the "brute force" parser for the recipe ingredients. 

How would it be if the parser replaces the unit with the desired notation e.g.:

100 g sugar -> 100 gr sugar
100 Gr sugar -> 100 gr sugar
100 gram sugar -> 100 gr sugar

For this the parser can use the abbreviation of the unit and replace it with the name of the unit. Or use a new field which contains the replacement. Depending on what is easier in programming. The user only has to add the appropriate replacements in the unit table. 

So everybody has his own wish how it should work or the choice if he wants to use it.

It would make it easier for me to import, parse and convert all my recipes, about 1000 pieces and also faster in the future. In addition, everything would always look the same.

**Comment by michael-genson** [comment]
How recipe ingredients are displayed is based on your Food/Unit stores now; the parser just tries to pick which food/unit the ingredient uses. You can use aliases to help the parser match different variations
